BootloaderCorePkg/Library/S3SaveRestoreLib/S3SaveRestore.c

```diff
@@ -95,8 +95,18 @@ AppendS3Info (
   if ( (UINT8 *)CommHdr + ((BL_PLD_COMM_HDR*)DataPtr)->TotalSize > SmmBase + SmmSize ) {
     return EFI_OUT_OF_RESOURCES;
   }
+
   CopyMem ((VOID *)CommHdr, DataPtr, ((BL_PLD_COMM_HDR*)DataPtr)->TotalSize);
 
+  //
+  // If the ID=SMMBASE_INFO_COMM_ID, then we are only copying the
+  // header and rest of TotalSize is placeholder for payload to fill in.
+  // So, do a ZeroMem on SmmBaseInfo data (TotalSize - sizeof(header)) 
+  //
+  if (CommHdr->Id == SMMBASE_INFO_COMM_ID) {
+    ZeroMem ((UINT8 *)CommHdr + sizeof(BL_PLD_COMM_HDR), ((BL_PLD_COMM_HDR*)DataPtr)->TotalSize - sizeof(BL_PLD_COMM_HDR));

```

Comment:
BTW, I saw many typecasting for (BL_PLD_COMM_HDR*)DataPtr.  So why don't we define a new variable  BL_PLD_COMM_HDR  *PldCommHdr to avoid these ?

Reply:
Sure. Let me change it.